types/types.go

```diff
@@ -445,8 +445,33 @@ type TraefikLog struct {
 
 // AccessLog holds the configuration settings for the access logger (middlewares/accesslog).
 type AccessLog struct {
-	FilePath string `json:"file,omitempty" description:"Access log file path. Stdout is used when omitted or empty" export:"true"`
-	Format   string `json:"format,omitempty" description:"Access log format: json | common" export:"true"`
+	FilePath         string           `json:"file,omitempty" description:"Access log file path. Stdout is used when omitted or empty" export:"true"`
+	Format           string           `json:"format,omitempty" description:"Access log format: json | common" export:"true"`
+	HeaderRedactions HeaderRedactions `json:"headerredactions,omitempty" description:"List of names of sensitive headers to redact" export:"true"`
+}
+
+// HeaderRedactions holds a list of HTTP header names
+type HeaderRedactions []string
+
+// Set splits string on , and adds the items to the list of headers to
+// be redacted
+func (b *HeaderRedactions) Set(str string) error {
+	headers := strings.Split(str, ",")
+	for _, header := range headers {
+		*b = append(*b, header)
```

Comment:
What happens of someone adds blanks around header names? Shouldn't we trim each header?

Reply:
What happens if someone uses an empty string (i.e., `Authorization,Secret,,Confidential`)? Shouldn't we skip empty values?